### PR TITLE
Implement entire CloudWatch API

### PR DESCRIPTION
**Pull request by fmcgeough**
This is a full implementation of the 2010-08-01 CloudWatch API
with doctests for functions and fairly complete doc. Modified
mix.exs for dev dependency on ex_doc. Modified CloudwatchTest to
add doctest line. Issue https://github.com/ex-aws/ex_aws_cloudwatch/issues/3

### CODE REVIEW

**Comment by cammellos** [comment]
Thanks for the PR, I will try to have a look at it this weekend!

**Review by cammellos — APPROVED** [review]
looks good to me, we might want to add some tests in the future, but that's a discussion for later.
thanks

File: lib/ex_aws/cloudwatch.ex [hunk 1]
```diff
@@ -1,42 +1,603 @@
...
+  publish data using a statistic set instead, you can only retrieve percentile
+  statistics for this data if one of the following conditions is true:
+
+  * The SampleCount value of the statistic set is 1
+  * The Min and the Max values of the statistic set are equal
+
+
+  """
+  @spec put_metric_data_input(metric_data :: [metric_datum, ...], namespace :: binary) :: ExAws.Operation.Query.t()
+  def put_metric_data_input(metric_data, namespace) do
```
Comment (cammellos) [inline lib/ex_aws/cloudwatch.ex]: Should this be put_metric_data instead?  https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_PutMetricData.html

**Review by cammellos — CHANGES_REQUESTED** [review]

File: lib/ex_aws/cloudwatch.ex (hunk 1 above)
Comment (fmcgeough) [reply lib/ex_aws/cloudwatch.ex]: shoot. you are absolutely correct. this was one method that I haven't used so I didn't notice that it was the wrong name. I've changed it.

**Review by cammellos — APPROVED** [review]

**Comment by CaldwellYSR** [comment]
Are there plans to merge this in? I would like to use it in my project but I don't really want to use the fork if that's not where the future updates will be.

**Comment by yumeto** [comment]
I'm waiting for this PR merged, too.

**Comment by cammellos** [comment]
@yumeto Thanks, I can't merge it myself as don't have write access, @benwilson512 do you have time to look at this?

**Comment by yumeto** [comment]
@cammellos Thanks, too. Actually I want only `put_metric_data` functionality proposed by #2, and I found this PR, which completely covers #2 with `typespecs`.

**Pull request merged**